### PR TITLE
Wrap inline hint style

### PR DESCRIPTION
**Pull request by dandavison**
@th1000s This PR (targeting your branch) adds a new option `inline-hint-style` and uses it to style the wrap and prefix symbols. The option accepts values using the same style mini-language as the other delta style options.

I haven't pushed this to your branch yet because your suggestion of using syntect [ThemeSettings](https://docs.rs/syntect/2.1.0/syntect/highlighting/struct.ThemeSettings.html) is interesting. Perhaps one consideration there is that some delta users do not use syntax highlighting (e.g. those using the diff-highlight / diff-so-fancy emulation modes, or the `villsau` delta theme.)

### CODE REVIEW

File: src/features/side_by_side_wrap.rs [hunk 1]
```diff
@@ -293,10 +294,23 @@ pub fn wrap_plusminus_block<'c: 'a, 'a>(
...
+        let inline_hint_style = if config
+            .inline_hint_style
+            .ansi_term_style
+            .background
+            .is_some()
+        {
+            Some(config.inline_hint_style)
+        } else {
+            None
+        };
```
Comment (dandavison) [inline src/features/side_by_side_wrap.rs:312]: I was expecting that, if `style.ansi_term_style.background = None`, then no change to the background color should be made. But in fact what I'm seeing is that the  background color is set to white for the symbol character when `style.ansi_term_style.background = None`. I haven't understood that yet.

My initial hope was not to use the `syntax_style_sections` at all, and instead only use the `diff_style_sections` (and in fact I have a [commit](https://github.com/dandavison/delta/commit/c5cde12e48a330c9b04a029e70987ce7f0185367) implementing that), but that wasn't possible due to this issue of the background color. In any case, this branch works and has an appropriate CLI, even if the implementation is a bit convoluted.

Comment (dandavison) [reply src/features/side_by_side_wrap.rs:312]: I think you're probably aware of this issue already, seeing as I'm following your code in using both `syntax_style_sections` and `diff_style_sections`!

Comment (th1000s) [reply src/features/side_by_side_wrap.rs:312]: Indeed, the interaction between these two colorings is quite something, good to know you also ran into that :)

**Comment by th1000s** [comment]
Ok, being able to set the `inline-hint-style` independent from syntax highlighting seems like a good idea.

**Comment by dandavison** [comment]
Thanks, I've merged this to your branch.

**Pull request merged**